### DISCUSSION TITLE
Is it possible to keep unused object properties when using `useFieldArray()`?

### DISCUSSION THREAD

**Discussion by mrlubos** [react-hook-form/react-hook-form#3341] [Q&A]
Hey! I am using `useFieldArray()` to manage a form, let's say the form state array looks like this.

```tsx
const fields = [{
  foo: 1,
  bar: 2,
}]
```

These values are managed by 2 separate controllers and all is working well. However, let's say my `defaultValues` looked like this.

```tsx
const fields = [{
  foo: 1,
  bar: 2,
  baz: 3,
}]
```

Now if I edit any of my controllers and submit the form, I will lose the `baz` property because it's not managed by any controller.

My question is if there's a way to keep the extra properties in my object? I don't want to render additional hidden controllers just so I can keep the state and merging states in `onSubmit()` doesn't feel ideal either (that's what I do now). Does anyone have any pointers? Thanks!

**Comment by bluebill1049** [comment]
I think we could make it work with `shouldUnreigster: false`, i will take a look at it.

**Reply by mrlubos** [reply]
Hi @bluebill1049, is that the recommended way or are further modifications needed (either on my end or react-hook-form)? I checked the documentation for this flag and I am a bit wary of losing the automatic form state updates. It feels like it invites potential bugs if a developer uses `shouldUnreigster: false` as it's not obvious they will lose form state in return. What do you think?

<img width="802" alt="Screenshot 2020-11-04 at 1 13 29 pm" src="https://user-images.githubusercontent.com/12529395/98110552-37a20180-1eda-11eb-86b0-60411bacd66b.png">

**Comment by bluebill1049** [comment]
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-2mxjs?file=/src/index.js
